### PR TITLE
chore: turn Step and Count into record

### PR DESCRIPTION
**Pull request by rajamatt**
Refactors Step and Count into an immutable record #https://github.com/unoplatform/private/issues/446

Updates Uno.Extensions.Reactive to 4.0.0 to be able to use UpdateAsync()

### CODE REVIEW

File: reference/Counter/CSharp-MVUX/Counter/MainModel.cs [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 internal partial record MainModel
 {
-    public IState<int> Count => State.Value(this, () => 0);
+    public IState<Countable> Countable => State.Value(this, () => new Countable(0, 1));
 
-    public IState<int> Step => State.Value(this, () => 1);
-    
-    public ValueTask IncrementCommand(int Step)
-            => Count.Update(c => c + Step, CancellationToken.None);
+    public ValueTask IncrementCommand()
```
Comment (kazo0) [inline reference/Counter/CSharp-MVUX/Counter/MainModel.cs:15]: @dr1rrb Would be curious if there's anything you see here that we should do differently.

File: reference/Counter/CSharp-MVUX/Directory.Packages.props [hunk 2]
```diff
@@ -1,16 +1,16 @@
 ﻿<Project ToolsVersion="15.0">
   <ItemGroup>
-    <PackageVersion Include="Microsoft.Extensions.Logging.Console" Version="8.0.0-rc.2.23479.6" />
+    <PackageVersion Include="Microsoft.Extensions.Logging.Console" Version="8.0.0" />
     <PackageVersion Include="Microsoft.Windows.Compatibility" Version="8.0.0-rc.2.23479.10" />
     <PackageVersion Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.756" />
-    <PackageVersion Include="Microsoft.WindowsAppSDK" Version="1.4.231008000" />
```
Comment (kazo0) [inline reference/Counter/CSharp-MVUX/Directory.Packages.props:6]: Are these package bumps needed for something?

Comment (rajamatt) [reply reference/Counter/CSharp-MVUX/Directory.Packages.props:6]: Uno.Extensions.Reactive 4.0.0 was needed for UpdateAsync, we only had Update before

**Review by kazo0 — APPROVED** [review]

**Review by agneszitte — APPROVED** [review]

**Review by ajpinedam — APPROVED** [review]
Looks good! Thanks

**Review by dr1rrb — APPROVED** [review]

**Pull request merged**